### DISCUSSION TITLE
Error when selecting model for Agent Builder

### DISCUSSION THREAD

**Discussion by yashsinha-aero** [danny-avila/LibreChat#4907] [Troubleshooting]
### What happened?

Tried creating an agent using agent builder, LC instance is hosted on HF, all models are modelspec-ed with a custom provider. When i try to select the model and provider for the agent and then click the “back” button, to go back to the agent builder I get the error pasted below. This makes creating agents impossible. Tried on multiple devices/browsers

### Steps to Reproduce

1. Create an Agent using Agent Builder
2. Use any model defined as a Model Spec ( i am not sure if it occurs for default models as well)
3. Clicking the back button arrow results in error

### Relevant log output

```shell
Unexpected Application Error!
n.toLowerCase is not a function. (In 'n.toLowerCase()', 'n.toLowerCase' is undefined)
i_@https://url.hf.space/assets/index.DQMzGBq6.js:3128:22888
A7@https://url.hf.space/assets/vendor.BujiSg_e.js:38:19558
Qee@https://url.hf.space/assets/vendor.BujiSg_e.js:40:44173
zee@https://url.hf.space/assets/vendor.BujiSg_e.js:40:39879
pSe@https://url.hf.space/assets/vendor.BujiSg_e.js:40:39802
v2@https://url.hf.space/assets/vendor.BujiSg_e.js:40:39651
j$@https://url.hf.space/assets/vendor.BujiSg_e.js:40:36006
iB@https://url.hf.space/assets/vendor.BujiSg_e.js:40:36811
Kf@https://url.hf.space/assets/vendor.BujiSg_e.js:38:3289
@https://url.hf.space/assets/vendor.BujiSg_e.js:40:34326
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
More details are required.

What are your modelSpecs configured like? (bare minimum reproduction)

What endpoints are selected for those modelSpecs? What is the list of available models for those endpoints?

**Reply by yashsinha-aero** [reply]
Sample of one of the ModelSpec. enforce+prioritize = true

      - name: "Instant-EP"
        label: "Instant"
        description: "Quickest model"
        preset:
          default: true
          endpoint: "myEP"
          model: "gpt-4o-mini"
          temperature: 0.90
          modelLabel: "Instant"
          greeting: |
            Hello! How can I help you today?
          promptPrefix: |
“””

The endpoint i am using is a LiteLLM one in which following models are added : "gpt-4o", "claude-3.5-sonnet", "gpt-4o-mini", "claude-sonnet-3.5", "chatgpt-4o-latest", "llama-3.2-90b-vision-preview", "gemini-exp-1114"
